storage/storage.go

```diff
@@ -131,15 +133,28 @@ func (cts *CTStorage) Add(ctx context.Context, entry *ctonly.Entry) (uint64, uin
 	defer span.End()
 
 	future := cts.storeData(ctx, entry)
-	idx, err := future()
-	if err != nil {
-		return 0, 0, fmt.Errorf("error waiting for Tessera future: %w", err)
-	}
-	if idx.IsDup {
-		return cts.dedupFuture(ctx, future)
-	}
-	return idx.Index, entry.Timestamp, nil
 
+	if cts.enableAwaiter {
+		idx, _, err := cts.awaiter.Await(ctx, future)
+		if err != nil {
+			return 0, 0, fmt.Errorf("error waiting for Tessera index future and its integration: %w", err)
+		}
+		if idx.IsDup {
```

Comment:
I think you can move this block outside of the if/else block.

Reply:
Moved the block to the upper level.